spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/route/support/AbstractRibbonCommand.java

```diff
@@ -109,13 +109,22 @@ protected static Setter getSetter(final String commandKey,
 		// @formatter:on
 	}
 
+	public boolean isExecuteWithLoadBalancer() {
+		return false;
+	}
+	
 	@Override
 	protected ClientHttpResponse run() throws Exception {
 		final RequestContext context = RequestContext.getCurrentContext();
 
 		RQ request = createRequest();
-		RS response = this.client.executeWithLoadBalancer(request, config);
-
+		RS response;
+		if (!isExecuteWithLoadBalancer()
```

Comment:
I am not understanding why only the `RestClientRibbonCommand` would call `execute` directly, can you explain?